cmd/kubeadm/test/cmd/join_test.go

```diff
@@ -46,3 +46,148 @@ func TestCmdJoinConfig(t *testing.T) {
 		kubeadmReset()
 	}
 }
+
+func TestCmdJoinDiscoveryFile(t *testing.T) {
+	if *kubeadmCmdSkip {
+		t.Log("kubeadm cmd tests being skipped")
+		t.Skip()
+	}
+
+	var initTest = []struct {
+		args     string
+		expected bool
+	}{
+		{"--discovery-file=foobar", false},
+		{"--discovery-file=file:wrong", false},
+	}
+
+	for _, rt := range initTest {
+		_, _, actual := RunCmd(*kubeadmPath, "join", rt.args, "--skip-preflight-checks")
+		if (actual == nil) != rt.expected {
+			t.Errorf(
+				"failed CmdJoinDiscoveryFile running 'kubeadm join %s' with an error: %v\n\texpected: %t\n\t  actual: %t",
+				rt.args,
+				actual,
+				rt.expected,
+				(actual == nil),
+			)
+		}
+		kubeadmReset()
+	}
+}
+
+func TestCmdJoinDiscoveryToken(t *testing.T) {
```

Comment:
Yes, but I just thought that it sounds like a little too much duplication really, what do you think?
They all test the same type of tokens anyway (e.g. foo isn't a valid token)